docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,181 @@
+frustrating experiences for users.
+
+### Goals
+
+- Once a day-2 host starts installation, the installation can be cancelled
+- There's a straight-forward way to trigger cancellation
+- Host whose installations are cancelled can be reclaimed back to their InfraEnvs and be reused
+- Host cancellation should not affect other hosts in the cluster and should not affect the cluster itself
+- Host cancellation should happen within a reasonable amount of time in order to support dynamic scaling 
+
+### Non-Goals
+
+- Hosts that are installed in a day-1 cluster is out of scope of this enhancement
+- Cluster installation cancellation is out of scope
+- The way reclaim/unbind works should not be affected by these changes
+- This enhancement is focused solely on Assisted-Service offerings using Kubernetes API (SaaS should not be affected by this)
+- The host should still be responsive, in other words, any hosts that are unreachable are out scope of this proposal
+
+## Proposal
+
+Allow cancelling an installing day-2 host through Assisted-Service's Kubernetes API.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, when I create a hosted cluster using Hypershift, I'd like to bind and install worker hosts into my cluster. 
+However, if I want to remove the host from the cluster before they finish installing, I should be able to do so and reuse the host for another cluster.
+
+#### Story 2
+
+As a cluster admin using Hypershift, I want to use the autoscaler.
+When the scaling happens and hosts are bound/unbound from the cluster, the hosts should bind and unbind 
+easily from the cluster.
+
+#### Story 3
+
+As a cluster admin, if I install a day-2 host to a cluster, and I see the host installation fails from
+inside the host. I should be able to cancel the installation once it starts and quickly reset the host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is an already existing cancel host installation API which maps to the unbind and reclaim stages.
+This is currently only being used when a day-1 cluster installation is cancelled. There is also
+existing logic in the 
+
+#### Detection of Cancellation
+
+When an Agent's cluster deployment reference is removed and the host is in the middle of installation,
+the controller will consider that a trigger to cancel the installation.
+
+Currently, there is a webhook that prevents the cluster reference from being removed
+when a host is in the middle of installation. The proposal for this enhancement is to remove
+that webhook to unblock the removal of the cluster reference.
+
+#### Cancelling Installation
+
+The host should transition to cancelled state. Once in cancelled state, the host can then
+be unbound from the cluster.
+
+#### Post Cancellation
+
+##### Successful Cancellation
+
+If a host cancellation was successful, the host will be transitioned to unbinding-pending-user-action.
```

Comment:
What does a successful cancellation mean? Does something need to be done on the host? 
Given the previous section says:
>Once in cancelled state, the host can then be unbound from the cluster.

What is the difference between cancelling and moving the host directly to unbinding-pending-user-action?

Reply:
Good question, modified the section to be automatic vs manual reset.

---
I initially thought cancel would send the `stop-installation` host command to the Agent, but it turns out not to be the case!